src/DiskIO/IpcIo/IpcIoFile.cc

```diff
@@ -372,6 +372,8 @@ IpcIoFile::push(IpcIoPendingRequest *const pending)
                dbName << " overflow: " <<
                SipcIo(KidIdentifier, ipcIo, diskId)); // TODO: report queue len
         // TODO: grow queue size
+        if (!pending->readRequest && ipcIo.page)
+            Ipc::Mem::PutPage(ipcIo.page); // Free the page if allocated
```

Comment:
Please remove the comment -- the code speaks (and speaks better) for itself in this case.